### PR TITLE
[editorconfig]: All md files except in test use tab

### PR DESCRIPTION
**Pull request by joshbruce**
**Marked version:** 0.3.17

## Description

tl;dr: PR changes editorconfig to use tabs not spaces while still ignoring MD files in test directory.

Looking at the CommonMark specification and considering the future regarding #1104 and https://github.com/markedjs/marked/issues/1103#issuecomment-369957173 a lot of the examples use honest tabs...not spaces in Markdown...makes sense given the nature of plain text editors (this was why @davisjam's initial contributions didn't pass lint - he uses VIM, which doesn't do spaced tabs).

So, not too sure what to do here regarding ignoring that `/test` folder. Or maybe ignoring markdown files all together like Federico had mentioned in the initial PR #1082 - think I forgot to put it back into WIP status or something and we hadn't really talked more about the GH Pages thing yet.

Plus, that's why we call it software and not hardware - can be easy to change in the future, if we do it right.

Thoughts?

Tagging @intcreator as well, as it's something to consider as we move toward creating all the tests ever.

## Contributor

- [ ] Test(s) exist to ensure functionality and minimize regresstion (if no tests added, list tests covering this PR); or,
- [x] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -13,5 +13,5 @@ end_of_line = lf
 insert_final_newline = true
 trim_trailing_whitespace = true
 translate_tabs_to_spaces = true
-indent_style = space
+indent_style = tab
 indent_size = 4
```
Comment (UziTech) [inline .editorconfig]: > when indent_style is set to tab, it may be desirable to leave indent_size unspecified so readers may view the file using their preferred indentation width

from http://editorconfig.org/#file-format-details

File: .editorconfig [hunk 2]
```diff
@@ -13,5 +13,5 @@ end_of_line = lf
 insert_final_newline = true
 trim_trailing_whitespace = true
 translate_tabs_to_spaces = true
```
Comment (UziTech) [inline .editorconfig]: What is this rule? it doesn't show on up https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties

Comment (joshbruce) [reply .editorconfig]: May be either a deprecated feature - or a Sublime Text specific thing (see first sentence after code): https://github.com/sindresorhus/editorconfig-sublime/issues/12

Automatically converts tabs to spaces on new files. Probably a horrible idea all around anyway. Good catch. ;)

Removed.

File: .editorconfig [hunk 3]
```diff
@@ -12,6 +12,5 @@ charset = utf-8
 end_of_line = lf
 insert_final_newline = true
```
Comment (UziTech) [inline .editorconfig]: kind of funny that this file doesn't have a new line at the end 😆

File: .editorconfig (hunk 1 above)
Comment (joshbruce) [reply .editorconfig]: Ran into issues with that, I think - most likely because of the asterisk. *all* files have an indent size of 2...need to override to say something else. We could remove the asterisk, I suppose. Basically saying we only want these specific file types to have these specific qualities. I tried doing something like this, and it didn't seem to work very well - but it could just be my editor + plug-in interaction (tested in both Sublime and VSC, with same unexpected results):

```
[*]
charset = utf-8
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true

[*.json, *.js]
indent_style = space
indent_size = 2

[*.md, !test/*md]
indent_style = tab
```

For some reason, when I tried various permutations of this inheritance stack I always got spaces in MD files...?? Could just be something in my environment, though it's not obvious to me what.

File: .editorconfig (hunk 3 above)
Comment (joshbruce) [reply .editorconfig]: Dude! I thought that was hilarious. I was literally expecting it to happen as soon as I hit safe...so disappointed. 🤣

File: .editorconfig (hunk 1 above)
Comment (joshbruce) [reply .editorconfig]: At one point, I was also tempted to just do this:

```
[*.json, *.js]
charset = utf-8
end_of_line = lf
insert_final_newline = true
indent_style = space
indent_size = 2
```

And call it a day...literally only effect JS and JSON. Not sure what would have things like YML though. Also not sure how broad our lint rules go either. Would like to get this to match those as much as possible where possible - just not a deal-breaker if they don't or can't.

**Comment by joshbruce** [comment]
Does eslint only do JS? (Revealing my over-dependance `editorconfig` right there.)

If so, could literally get rid of the other rules and just apply them to JS files. The objective is keep developers from failing lint, if their editor can apply the config. Give them the check with the `NPM lint:check` by @UziTech and a quick way to fix by Federico. And a way to protect the core by indicating with Travis. At that point, we copy-paste from the docs and point them there. Eventually people see it enough and should reduce how often it happens. :)

**Comment by joshbruce** [comment]
es = EcmaScript

I'm retarded sometimes... https://youtu.be/j9ndS6yu3Kk

**Comment by UziTech** [comment]
by default eslint only lints .js files. You can specify a [`--ext`](https://eslint.org/docs/user-guide/configuring#specifying-file-extensions-to-lint) parameter to add other extensions. `.json` is a pretty common one that is added
